lib/associations/belongs-to-many.js

```diff
@@ -113,7 +113,7 @@ var BelongsToMany = function(source, target, options) {
     this.foreignKeyAttribute = {};
     this.foreignKey = this.options.foreignKey || Utils.camelizeIf(
       [
-        Utils.underscoredIf(this.source.options.name.singular, this.source.options.underscored),
+        Utils.underscoredIf(this.options.as || this.source.options.name.singular, this.source.options.underscored),
```

Comment:
Should this be the singular name of the alias? `WorkersId` is pretty ugly, but we could run into a lot of issues like we've done previously.

@janmeier